### ISSUE TITLE
Stuck on "Downloading subtitle from PMS"

### ISSUE THREAD

**Issue by Skedooesh** [Hellowlol/bw_plex#38]
Hey! I am somehow still getting stuck when it's trying to download subtitle from PMS.

![image](https://user-images.githubusercontent.com/10135312/40666410-ea2975ce-635f-11e8-9dcd-e65bdf2f777f.png)

I have been trying to understand more about how this thing works lately, and I got everything else to work fine, with the token and url in the config file and etc. Would like to finally get this part to work now, if possible. 

I did `pip install -r requirements.txt` and it looked like it was downloading a lot of stuff, but I guess something must still be missing since it's stuck on this part.

If I run pip list, this is what I currently have installed. I tried some random stuff like pip install srt and such, but I have no idea how to locate the required one. Or maybe the issue lies somewhere else? Just did a full reinstall with the [all] option and removed everything from my .config/bw_plex folder before trying this. 

<details>
  <summary>List output: expand</summary>

- asn1crypto (0.24.0)
- audioread (2.1.5)
- backports.functools-lru-cache (1.5)
- beautifulsoup4 (4.6.0)
- bw-plex (0.0.8.dev10+g1362255, /home/stigus/Documents/plexShowSkip/bw_plex)
- certifi (2018.4.16)
- chardet (3.0.4)
- click (6.7)
- configobj (5.0.6)
- cryptography (2.1.4)
- cycler (0.10.0)
- decorator (4.3.0)
- docopt (0.6.2)
- download (0.3.3)
- dtrx (7.1)
- enum34 (1.1.6)
- et-xmlfile (1.0.1)
- f.lux-indicator-applet (1.1.11-pre)
- funcsigs (1.0.2)
- html5lib (1.0.1)
- idna (2.6)
- ipaddress (1.0.17)
- jdcal (1.4)
- joblib (0.11)
- keyring (10.6.0)
- keyrings.alt (3.0)
- kiwisolver (1.0.1)
- librosa (0.6.1)
- llvmlite (0.23.0)
- matplotlib (2.2.2)
- numba (0.38.0)
- numpy (1.14.3)
- odfpy (1.3.6)
- opencv-contrib-python (3.4.1.15)
- openpyxl (2.5.3)
- pexpect (4.2.1)
- Pillow (5.1.0)
- pip (9.0.1)
- PlexAPI (3.0.6)
- pocketsphinx (0.1.3)
- profilehooks (1.10.0)
- psutil (5.4.5)
- pycairo (1.16.2)
- pycrypto (2.6.1)
- pygobject (3.26.1)
- pyparsing (2.2.0)
- pytesseract (0.2.0)
- python-dateutil (2.7.3)
- pytz (2018.4)
- pyudev (0.21.0)
- pyxdg (0.25)
- PyYAML (3.12)
- requests (2.18.4)
- resampy (0.2.0)
- scikit-learn (0.19.1)
- scipy (1.1.0)
- SecretStorage (2.3.1)
- setuptools (39.2.0)
- singledispatch (3.4.0.3)
- six (1.11.0)
- SpeechRecognition (3.8.1)
- SQLAlchemy (1.2.8)
- srt (1.6.0)
- subprocess32 (3.5.1)
- tablib (0.12.1)
- tqdm (4.23.4)
- unicodecsv (0.14.1)
- urllib3 (1.22)
- webencodings (0.5.1)
- websocket-client (0.48.0)
- wheel (0.30.0)
- wsaccel (0.6.2)
- xlrd (1.1.0)
- xlwt (1.3.0)
- youtube-dl (2018.5.26)

</details>

I don't really know where to troubleshoot anymore, as everything else seems to work just fine.

**Comment by Skedooesh** [comment]
It might actually be stuck somewhere else, previously I was able to remove the words from config.ini and it would jump past the download etc, but now it's stuck even if I do that. 

![image](https://user-images.githubusercontent.com/10135312/40669908-4c7852e2-6368-11e8-9333-533acefb25ae.png)

When I did that before it would just continue, doing the episode progress check every 8 ish second, and skipping when it was in the right time interval.

I left it like that for 5+ minutes as of right now, and nothing seems to happen. 

Starting a new episode does pull it out of whatever is happening, until it gets to the same point on the new episode. 
![image](https://user-images.githubusercontent.com/10135312/40670236-1e52d152-6369-11e8-95f2-0c56dc2968f1.png)

**Comment by Hellowlol** [comment]
Crap. My bad. Ill fix that.

**Comment by Hellowlol** [comment]
git pull should fix that for you. Ill add some more addition logging and a proper exception handler later today (so errors isnt swallowed by ```watch```)

**Comment by Skedooesh** [comment]
Allright, it's working better after that fix, but now it's stuck in a loop it seems. process works perfectly without any errors, but when using watch, it loops the current and next episode over and over. So it does everything for the current episode -> next episode -> current... and keeps going like that. So it never gets to the actual skipping part atm.

**Comment by Skedooesh** [comment]
Yeah, guess it might be bad news sharing that lol. Thanks for fixing. Did you get what you needed though, or should I post a log where I perform another command?

**Comment by Hellowlol** [comment]
Hmm, i just think it strange that it logs that that westworld s01e01 was added to the db, but on the next run, it says it fail to find it. Can you check the db manually? 

After that wipe hashes.pklz the log and media.db, use the watch command let it run for like 5 minutes, control c then poste the log after sanitation! and your config file (remember to remove stuff from there too, just write REMOVED or something on username, password and token)  post it to a gist so i dont have to download it :)

**Comment by Hellowlol** [comment]
Also what client are you using?

**Comment by Hellowlol** [comment]
I just want to see what settings you have disabled, it quick to disable something you dont want to. url and token is enough but remove the token before you post it here xD.

**Comment by Skedooesh** [comment]
Haha, fair enough ^^,

https://gist.github.com/Skedooesh/65803a2f4461035c1d900959042a20f8

**Comment by Skedooesh** [comment]
Fresh log after cleaning up everything: https://gist.github.com/Skedooesh/edd5998d877d130eaa6d92e6b10ee9b2

My media.db now looks like this:

![image](https://user-images.githubusercontent.com/10135312/40679780-6b5196c6-6384-11e8-8f51-4b5adb6ed7dd.png)

**Comment by Hellowlol** [comment]
Thanks its a bug in my sql. Im not that familiar with that. Thanks for the report.

**Comment by Hellowlol** [comment]
Its "fixed" in the lastest commit. You will need to delete the db before you start the program. Also can you enable check_credits

also can you change your config ini so you values are the same as this?:
```
[tv]
check_credits_action = 
check_credits_start_next_ep = True
check_credits_sec = 120
check_for_theme_sec = 600
check_intro_ffmpeg_sec = 600

[movie]
check_credits = True
check_credits_action = stop
check_credits_sec = 600
check_intro_ffmpeg_sec = 600
```
That way you can test more features :)

**Comment by Skedooesh** [comment]
Awesome, it's working again! :) Sure, i'll test it out and report any oddities. Also, do you have a donate link anywhere? You are on point with the support here, would be nice to contribute a little.

**Comment by Hellowlol** [comment]
I didn’t plan setting up a donation link before this was really usageble. But my paypal adresse is hellowlol1@gmail.com

I’ll fix the name error. I also expect that some of the other commands broke when I added movie support.

**Comment by Skedooesh** [comment]
It's coming along nicely, will contribute a bit. 
Yeah, I noticed some of the other commands had the same error :)

**Comment by Hellowlol** [comment]
Can you verify that the credits bug is fixed?

**Comment by Skedooesh** [comment]
Yeah, works correctly now, used jump in theme interval and then started to process a new episode when that was done 👌  Process command instead of watch also worked perfect :)
